File: java101/Alapok/2017.10.14 stringek/palindroma/src/palindroma/Palindroma.java
package palindroma;
import java.util.Scanner;
public class Palindroma {
    
    static boolean vizsgalat(String s) {
        int bal = 0;
        int jobb = s.length()-1;
        
        while (bal < jobb) {            
            if (s.charAt(bal) != s.charAt(jobb))
                return false;
        bal++;
        jobb--;
        }
        return true;
    }
    
    public static void main(String[] args) {
        Scanner bill = new Scanner(System.in, "Cp1250");
        System.out.print("Szöveg: ");
        String be = bill.nextLine().toUpperCase().replace(" ", "");
        System.out.println("Átalakítva: " + be);
        if (vizsgalat(be)) {
            System.out.println("Palindróma!");
        } else {
            System.out.println("Nem palindróma!");
        }
    }
}

File: python101/python2/11 Stringek 2/Megoldások/palindroma.py
# Palindróma

while True:
    be = input('Írd be a szöveget: ')
    if be == '':
        break
    at = be.upper().replace(' ','')
    print('Átalakítva:',at)
    ford = ''
    for c in at:
        ford = c + ford
    print('Fordítva:',ford)
    if ford == at:
        print('Palindróma')
    else:
        print('Nem palindróma')

